src/cascadia/TerminalAzBridge/ConsoleInputReader.cpp

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "ConsoleInputReader.h"
+
+ConsoleInputReader::ConsoleInputReader(HANDLE handle) :
+    _handle(handle)
+{
+    _buffer.resize(BufferSize);
+    _convertedString.reserve(BufferSize);
+}
+
+void ConsoleInputReader::SetWindowSizeChangedCallback(std::function<void()> callback)
+{
+    _windowSizeChangedCallback = std::move(callback);
+}
+
+std::optional<std::wstring_view> ConsoleInputReader::Read()
+{
+    DWORD readCount{ 0 };
+
+    _convertedString.clear();
+    while (_convertedString.empty())
+    {
+        _buffer.resize(BufferSize);
+        BOOL succeeded =
+            ReadConsoleInputW(_handle, _buffer.data(), gsl::narrow_cast<DWORD>(_buffer.size()), &readCount);
+        if (!succeeded)
+        {
+            return std::nullopt;
+        }
+
+        _buffer.resize(readCount);
+        for (auto it = _buffer.begin(); it != _buffer.end(); it++)

```

Comment:
```suggestion
        for (auto it = _buffer.begin(); it != _buffer.end(); ++it)
```

To avoid the copy.

Reply:
Right, thanks. I was troubleshooting a "lol it doesn't do `operator++`" that turned out to be because i had it  `const`